libraries/AP_HAL_Linux/SPIUARTDriver.cpp

```diff
@@ -194,7 +136,7 @@ int SPIUARTDriver::_read_fd(uint8_t *buf, uint16_t n)
     _dev->transfer_fullduplex(ff_stub, buf, n);
     _dev->get_semaphore()->give();
 
-    BUF_ADVANCETAIL(_readbuf, n);
+    _readbuf.commit(n);
```

Comment:
this commit() alone here is pretty confusing (and probably wrong since UARTDriver::_read_fd() doesn't advance the ByteBuffer. Can we put the commit() call together with the caller or at least a comment above this call if it is correct?  I don't think it is correct since with a quick check this is called from UARTDriver::_timer_tick(), which already advances the pointer.